internal/storage/v1/elasticsearch/spanstore/index_utils.go

```diff
@@ -7,8 +7,70 @@ import (
 	"time"
 )
 
+type TimeRangeIndexFn func(indexName string, indexDateLayout string, startTime time.Time, endTime time.Time, reduceDuration time.Duration) []string
+
+func TimeRangeIndicesFn(useReadWriteAliases bool, readAliasSuffix string, remoteReadClusters []string) TimeRangeIndexFn {
```

Comment:
can we just keep everything in the same reader file? It's easier to review, no value in separating.